### ISSUE TITLE
Menu link search system

### ISSUE THREAD

**Issue by micker** [joomla-projects/j4adminui#23]
hello select list for menu link is realy a good feature but need to add search with dynamic filter ... it will be perfect
![image](https://user-images.githubusercontent.com/2539440/68158184-685dfc80-ff4f-11e9-9670-e507fcbac92c.png)

**Comment by ahamed** [comment]
@micker, Could you please explain what kind of **dynamic filtering** have you been asking?

**Comment by micker** [comment]
hello
like choice.js before article level a search:
tip what you want and list become filtered
like blog => link to category with blog ...
you understand ?

**Comment by brianteeman** [comment]
Is this list approach practical on a site with 20 components installed?

**Comment by micker** [comment]
i thnk with a search system it will be an interesting workflow

**Comment by imarklee** [comment]
> Is this list approach practical on a site with 20 components installed?

If an extension has a lot of layouts, it could be problematic too.

**Comment by ahamed** [comment]
I understood the **search** functionality but what about the **dynamic filter**? We can add a search field at the top of the list and search from the list.

Could you explain the **dynamic filter** with a proper example?

**Comment by micker** [comment]
sorry for dynamic filter the idea is to display good menu link according to words tips

@imarklee maybe but its the same problem with a site with 30 category 
as i can test workflow seams better with this list because :
- no more other page for user
- better in mobile device

**Comment by brianteeman** [comment]
> > Is this list approach practical on a site with 20 components installed?

> If an extension has a lot of layouts, it could be problematic too.

I think its a nice idea but not practical in the real world - people can and do install a lot of extensions

**Comment by micker** [comment]
the modal with 20 components isn't good too not ?

**Comment by imarklee** [comment]
> sorry for dynamic filter the idea is to display good menu link according to words tips
> 
> @imarklee maybe but its the same problem with a site with 30 category
> as i can test workflow seams better with this list because :
> 
>     * no more other page for user
> 
>     * better in mobile device

Yep, I love this idea and this is a very elegant solution. With a search just like select2, it would be easier but currently the layout could bleed if the site has many extensions or layouts in an extension. 

Example:

<img width="960" alt="menu_list" src="https://user-images.githubusercontent.com/892204/69531122-4683f280-0fae-11ea-9b84-8ca346118916.png">

It's very common for users to have more than 30 - 50 extensions since Joomla is so modular.

**Comment by imarklee** [comment]
Perhaps a fixed height + scrollbar could solve this?

**Comment by micker** [comment]
yep +10
